fluidBuild.config.cjs

```diff
@@ -578,10 +578,6 @@ module.exports = {
 				// 	name: "api",
 				// 	body: "fluid-build . --task api",
 				// },
-				{
-					name: "build:docs",
-					body: "api-extractor run --local",
-				},
```

Comment:
Are there packages that had a policy exception that could then(now) be removed?

Reply:
If we're going to remove this one we should remove the ci:build:docs entry too.

Reply:
The current policy changes helps remove all the exclusions listed in the fluid build config: https://github.com/microsoft/FluidFramework/blob/main/fluidBuild.config.cjs#L447. To remove the exclusions, a build-tools release is needed.

Reply:
I recommend against making the npm-public-package-requirements policy more complicated. Its purpose is to enable strong enforcement, and I think it's fine to leave that capability as it is. If these scripts aren't required in all our packages, then they shouldn't be in an entry in those settings. Just remove the build:docs and ci:build:docs entries in settings. If anything, we should make the list of "prod packages" configurable, because then that logic doesn't live in code that has to be updated and released when we need to change it.

You can also remove the requiredDevDependencies setting because we already have a policy that checks for dependencies if you have a script that uses a command from that dep. See the commandPackages property in the fluidBuild config and the npm-package-json-script-dep policy. I just tested it on a package that doesn't have an api-extractor dep. If you add a script that calls api-extractor, the policy will start to fail.

If we are also trying to enforce something about packages that have exports, I recommend a new policy that does just the necessary check. Policies that are simple and do one thing are much easier to work with. It is a fine pattern to have a policy that applies to most but not all of the projects and just exclude the projects where it doesn't apply. But when policies are complex and check a lot of things, then we can't disable them as easily, because disabling one piece of its functionality disables all of it. Smaller, "dumber" policies are better.

Reply:
example of what I mean: https://github.com/microsoft/FluidFramework/commit/63ab3ad28253d041a41c77190f4625e41f8d506c